src/iuse_actor.cpp

```diff
@@ -694,6 +694,7 @@ long pick_lock_actor::use( player *p, item *it, bool, const tripoint& ) const
         it->damage++;
         if( it->damage >= 5 ) {
             p->add_msg_if_player( m_bad, _( "The lock stumps your efforts to pick it, and you destroy your tool." ) );
+            p->i_rem(it);
```

Comment:
Removing it here means that the `it` pointer is invalidated.
This means that the weird check below (`it->damage < 100`? When will it be anything but true anyway?) invokes unspecified behavior, which could lead to crashes or at the very least random results.